### PR TITLE
Trev8939/stretch renderer

### PR DESCRIPTION
**Pull request by TADraeseke**
Adds a sample demonstrating use of the StretchRenderer.

### CODE REVIEW

**Comment by TADraeseke** [comment]
https://devtopia.esri.com/runtime/java-android-sdk/issues/1303

File: stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.widget.Toast;
+import com.esri.arcgisruntime.layers.RasterLayer;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.raster.*;
```
Comment (doneill) [inline stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/MainActivity.java]: Import classes explicitly.

File: stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/ParametersDialogFragment.java [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.DialogFragment;
+import android.support.v7.app.AlertDialog;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.*;
```
Comment (doneill) [inline stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/ParametersDialogFragment.java]: Import classes explicitly.

File: stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/ParametersDialogFragment.java [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+        updateSeekBar(mStdDevSeekBar, mStdDevFactor, mCurrStdDevTextView);
+      }
+
+      @Override
+      public void onStartTrackingTouch(SeekBar seekBar) { }
+
+      @Override
+      public void onStopTrackingTouch(SeekBar seekBar) { }
+    });
+    // stretch type spinnner
```
Comment (doneill) [inline stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/ParametersDialogFragment.java]: sp extra `n`

File: stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/ParametersDialogFragment.java [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+  @SuppressLint("SetTextI18n")
+  private void updateSeekBar(SeekBar seekBar, Integer progress, TextView textView) {
+    seekBar.setProgress(progress);
+    textView.setText(progress.toString());
+  }
+
+  /**
+   * Interface for passing dialog parameters back to MainActivity.
+   */
+  public interface ParametersListener {
```
Comment (doneill) [inline stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/ParametersDialogFragment.java]: Access can be package private

File: stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    mMapView.pause();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mMapView.resume();
+  }
+
+  public enum StretchType {
```
Comment (doneill) [inline stretch-renderer/src/main/java/com/esri/arcgisruntime/sample/stretchrenderer/MainActivity.java]: Access can be package-private

File: stretch-renderer/src/main/res/layout/dialog_box.xml [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+            android:layout_marginTop="15dp"
+            app:layout_constraintTop_toBottomOf="@+id/min_value_text_view"
+            android:layout_marginLeft="16dp"
+            app:layout_constraintLeft_toLeftOf="parent"
+            android:layout_marginStart="16dp"/>
+    <SeekBar
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:id="@+id/min_seek_bar"
+            android:layout_marginLeft="0dp"
```
Comment (doneill) [inline stretch-renderer/src/main/res/layout/dialog_box.xml]: Refactor to `android:layout_marginStart`

File: stretch-renderer/src/main/res/layout/dialog_box.xml [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+            android:layout_marginTop="15dp"
+            app:layout_constraintTop_toBottomOf="@+id/percent_clip_min_value_text_view"
+            android:layout_marginLeft="16dp"
+            app:layout_constraintLeft_toLeftOf="parent"
+            android:layout_marginStart="16dp"/>
+    <SeekBar
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:id="@+id/percent_clip_min_seek_bar"
+            android:layout_marginLeft="0dp"
```
Comment (doneill) [inline stretch-renderer/src/main/res/layout/dialog_box.xml]: Refactor to `android:layout_marginStart`

File: stretch-renderer/src/main/res/layout/dialog_box.xml [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+            app:layout_constraintTop_toTopOf="@+id/percent_clip_max_value_text_view"
+            android:layout_marginEnd="16dp"/>
+
+    <!-- StandardDeviation -->
+    <TextView
+            android:text="@string/standard_deviation"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:id="@+id/std_dev_text_view"
+            android:layout_marginLeft="16dp"
```
Comment (doneill) [inline stretch-renderer/src/main/res/layout/dialog_box.xml]: Refactor to `android:layout_marginStart`

File: stretch-renderer/src/main/res/layout/dialog_box.xml [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+            android:visibility="gone"/>
+    <SeekBar
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:id="@+id/std_dev_seek_bar"
+            app:layout_constraintTop_toTopOf="@+id/std_dev_text_view"
+            android:layout_marginTop="0dp"
+            app:layout_constraintBottom_toBottomOf="@+id/std_dev_text_view"
+            android:layout_marginBottom="0dp"
+            android:layout_marginLeft="0dp"
```
Comment (doneill) [inline stretch-renderer/src/main/res/layout/dialog_box.xml]: Refactor to `android:layout_marginStart`

File: stretch-renderer/src/main/res/layout/dialog_box.xml [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+            android:layout_marginLeft="16dp"
+            app:layout_constraintLeft_toLeftOf="parent"
+            android:layout_marginStart="16dp"/>
+    <SeekBar
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:id="@+id/min_seek_bar"
+            android:layout_marginLeft="0dp"
+            app:layout_constraintRight_toLeftOf="@+id/curr_min_text_view"
+            android:layout_marginRight="8dp"
```
Comment (doneill) [inline stretch-renderer/src/main/res/layout/dialog_box.xml]: Refactor to `android:layout_marginEnd`

File: stretch-renderer/src/main/res/layout/dialog_box.xml [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+            android:layout_marginLeft="16dp"
+            app:layout_constraintLeft_toLeftOf="parent"
+            android:layout_marginStart="16dp"/>
+    <SeekBar
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:id="@+id/percent_clip_min_seek_bar"
+            android:layout_marginLeft="0dp"
+            app:layout_constraintRight_toLeftOf="@+id/curr_percent_clip_min_text_view"
+            android:layout_marginRight="8dp"
```
Comment (doneill) [inline stretch-renderer/src/main/res/layout/dialog_box.xml]: Refactor to `android:layout_marginEnd`

File: stretch-renderer/src/main/res/layout/dialog_box.xml [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:id="@+id/std_dev_seek_bar"
+            app:layout_constraintTop_toTopOf="@+id/std_dev_text_view"
+            android:layout_marginTop="0dp"
+            app:layout_constraintBottom_toBottomOf="@+id/std_dev_text_view"
+            android:layout_marginBottom="0dp"
+            android:layout_marginLeft="0dp"
+            app:layout_constraintRight_toLeftOf="@+id/curr_std_dev_text_view"
+            android:layout_marginRight="16dp"
```
Comment (doneill) [inline stretch-renderer/src/main/res/layout/dialog_box.xml]: Refactor to `android:layout_marginEnd`

**Comment by doneill** [comment]
@TADraeseke Need to add module to settings.gradle

**Comment by TADraeseke** [comment]
@doneill All proposed changes made. Back to you!

**Comment by doneill** [comment]
@TADraeseke Updates look good to merge.

**Pull request merged**